### ISSUE TITLE
feature request: warn discarding exit code of `test`, `[`, and `[[`

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3031]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
(
  test $# -gt 0
# ^-----------^ SCXXXX (level): Do not ignore exit status of test.
  if test $# -gt 0; then return; fi  # OK
)

(
  [ $# -gt 0 ]
# ^----------^ SCXXXX (level): Do not ignore exit status of [.
  [ $# -gt 0 ] || exit  # OK
  [ $# -gt 0 ] && exit  # OK
)

(
  [[ $# -gt 0 ]]
# ^------------^  SCXXXX (level): Do not ignore exit status of [[.
  set -e
  [[ $# -gt 0 ]]  # OK
)
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

`SCXXXX (level): Do not ignore exit status of test/[/[[.`

**Comment by wileyhy** [comment]
Hm... The presence of `set -e` in a script could indicate that it may have
been intended for execution to have stopped whenever an exit status of
test/[/[[ had been ignored.
Wiley

**Comment by plambert** [comment]
Yes, shellcheck is aware of and tracks `set -e`'s status, which is why in the third example @e-kwsm doesn't show an error message after the `set -e`.

**Comment by jsmucr** [comment]
If I may add something here, the return code of `(( <expr> ))` should be checked too. Consider this:
```bash
set -e
I=0
(( I++ ))
echo 'This is not printed.'
```
